src/modal/modal.service.ts

```diff
@@ -67,6 +68,33 @@ export class ModalService {
 		return component;
 	}
 
+	/**
+	 * Creates and renders a new alert modal component.
+	 * @param data You can pass in `title`, `text` and `buttons` to be used in the modal.
+	 * `buttons` is an array of objects
+	 * ```
+	 * {
+	 * 		text: "Button text",
+	 * 		type: "primary" | "secondary" | "tertiary" | "ghost" | "danger" | "danger--primary" = "primary",
+	 * 		click: clickFunction,
+	 * }
+	 * ```
+	 * @returns {ComponentRef<any>}
+	 * @memberof ModalService
+	 */
+	show(data: {modalType?: string, headerText?: string, title: string, text: string, buttons?: null}) {
```

Comment:
Nice

Reply:
Thank you, I just followed your idea and expanded as much as we need it in our project :)